functions/nebari.js

```diff
@@ -0,0 +1,102 @@
+import {
+  pushToGitHub,
+  getZip_Uid,
+  getRootUrlWithoutTrailingSlash
+} from './utils'
+
+const repoOwner = process.env.VUE_APP_GH_USER
+const repo = process.env.VUE_APP_GH_REPO
+
+// This function is the one Netlify function runs on
+// https://docs.netlify.com/functions/build-with-javascript/#synchronous-function-format
+exports.handler = async function (event, _) {
+  // event is a JSON object
+  const data = JSON.parse(event.body)
+  const template = `ignite-${data.template}`
+  const nebariInstanceLink = getRootUrlWithoutTrailingSlash(
+    data.nebariInstanceLink
+  )
+  const userName = data.userName
+  const nbName = `${template}.ipynb`
+  const { zipRes, nbUid } = await getZip_Uid(data)
+
+  const title = template
+    .replace('ignite-', '')
+    .split('-')
+    .map((v) => v[0].toUpperCase() + v.slice(1))
+    .join(' ')
+  // notebook cell structure
+
+  function create_nb_cell(source_array, cell_type) {
+    return {
+      cell_type: cell_type,
+      metadata: {},
+      execution_count: null,
+      outputs: [],
+      source: source_array
+    }
+  }
+
+  let specific_commands = []
+
+  if (title === 'Template Vision Segmentation') {
+    specific_commands.push(
+      '!python -c "from data import download_datasets; download_datasets(\'./\')"'
+    )
+  }
+
+  const md_cell = [
+    `# ${title} by PyTorch-Ignite Code-Generator\n\n`,
+    'Please, run the cell below to execute your code.'
+  ]
+
+  const common_nb_commands = [
+    `!wget ${zipRes}\n`,
+    `!unzip ${template}.zip\n`,
+    '!pip install -r requirements.txt'
+  ]
+
+  const argparser = data.argparser
+  const execution_nb_commands = [
+    `!python main.py ${
+      argparser === 'hydra'
+        ? '#--config-dir=/content/ --config-name=config.yaml'
+        : 'config.yaml'
+    }`
+  ]
```

Comment:
Can we refactor these methods with those from colab.js to make less code duplications?